### PR TITLE
Fix/git-commands-cross-platform

### PR DESCRIPTION
**Pull request by andrewalderson**
This PR addresses issues with the git commands not working cross-platform.
It also address some issues that make development of this library a little more stable
by linking the source directory as a project dependency ensuring that everything
continues to work when commiting changes.

### CODE REVIEW

**Comment by LMaxence** [comment]
the way we run git commands is something I truly truly fear ! Let me just run some tests on my side today and I'll be more than happy to approve this :)

Many thanks for your work making Mookme compatible with Windows !

File: package.json [hunk 1]
```diff
@@ -23,8 +23,8 @@
   },
   "homepage": "https://github.com/Escape-Technologies/mookme",
   "devDependencies": {
-    "@commitlint/cli": "^12.1.1",
-    "@commitlint/config-angular": "^12.1.1",
-    "@escape.tech/mookme": "^1.2.0-beta.7"
+    "@commitlint/cli": "^17.0.0",
+    "@commitlint/config-angular": "^17.0.0",
+    "@escape.tech/mookme": "file:packages/mookme"
```
[the remark this answers is not in the document]
Comment (LMaxence) [reply package.json:28]: Thanks for this ! I learnt something today

**Comment by LMaxence** [comment]
This seems good to me :) Just a quick question before I merge it

File: .hooks/commit-msg.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
-    "steps": [{
-        "name": "commit lint",
-        "command": "cat {args} | ./node_modules/@commitlint/cli/cli.js"
-    }]
-}
\ No newline at end of file
+  "steps": [
+    {
+      "name": "commit lint",
+      "command": "npx commitlint --edit"
```
[the remark this answers is not in the document]
Comment (LMaxence) [reply .hooks/commit-msg.json:5]: Anyway I tested it and things seem fine, I'll trust you on this one

**Comment by LMaxence** [comment]
Deployed with `2.2.0-beta.4` :)

File: .hooks/commit-msg.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewalderson) [reply .hooks/commit-msg.json:5]: I am not an expert on this stuff but here is what the commitlint cli docs say
```
 -e, --edit           read last commit message from the specified file or
                       fallbacks to ./.git/COMMIT_EDITMSG               [string]
```
You can pass the file path as an argument but I don't understand why that is necessary.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewalderson) [reply package.json:28]: With npm packages you can install a local ```package.json``` file. When creating typescript/javascript libraries you should create a test/examples project and install your local dev enviroment. You can then build your local project in watch mode while you do dev.

**Pull request merged**